src/conn/conn_ckpt.c

```diff
@@ -25,6 +25,12 @@ __ckpt_server_config(WT_SESSION_IMPL *session, const char **cfg, bool *startp)
 
     conn = S2C(session);
 
+    WT_RET(__wt_config_gets(session, cfg, "checkpoint.precise_checkpoint", &cval));
+    if (cval.val)
+        F_SET(conn, WT_CONN_PRECISE_CHECKPOINT);
+    else
+        F_CLR(conn, WT_CONN_PRECISE_CHECKPOINT);
+
```

Comment:
The "always do a precise checkpoint if we're disagg" functionality isn't in this PR - is that in a separate ticket?

Reply:
Added